### ISSUE TITLE
[V4] `postcss-preset-env` and `autoprefixer` not loaded

### ISSUE THREAD

**Issue by stijns96** [tailwindlabs/tailwindcss#15516]
**What version of Tailwind CSS are you using?**
v4.0.0-beta.8

**What build tool (or framework if it abstracts the build tool) are you using?**

For example: postcss-cli 8.3.1, Next.js 10.0.9, webpack 5.28.0
- postcss 8.4.49
- postcss-load-config 6.0.1

**What version of Node.js are you using?**

v20.10.0

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

A Tailwind Play link or public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways.

A reproduction is **required** when filing an issue — any issue opened without a reproduction will be closed and you'll be asked to create a new issue that includes a reproduction. We're a small team and we can't keep up with the volume of issues we receive if we need to reproduce each issue from scratch ourselves.

**Describe your issue**

I'm creating a small CLI that is basically handle my css files at this point. I followed the v4 docs and added the dependencies and created a small script

**build.mjs**
```js
import { basename } from 'path';
import fse from 'fs-extra';
import fg from 'fast-glob';
import postcss from 'postcss';
import postcssrc from 'postcss-load-config';

export default class Build {
  async run() {
    await this.buildCss();
  }

  async buildCss() {
    const input = fg.globSync('./src/assets/**/*.css');
    const { plugins } = await postcssrc();
    // Process files sequentially to respect potential dependencies
    for (const path of input) {
      const fileName = basename(path);
      const css = fse.readFileSync(path, { encoding: 'utf-8' });
      const result = await postcss(plugins).process(css, {
        from: path,
      });
      await fse.outputFile(`./dist/assets/${fileName}`, result.css);
    }
  }
}
```

**postcss.config.mjs**
```js
export default {
  plugins: {
    '@tailwindcss/postcss': {},
  },
};
```

Somehow `postcss-preset-env` and `autoprefixer` are not working, while the docs say I can remove it. By adding `postcss-preset-env`, both work nicely. `autoprefixer` is already implemented in `postcss-preset-env`

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/v4-beta#built-in-css-transpilation), v4 indeed does not use `postcss-preset-env` and `autoprefixer` but rather it relies on Lightning CSS:

https://github.com/tailwindlabs/tailwindcss/blob/608b1399cd66ec391846a81a154a6593db64b040/packages/%40tailwindcss-postcss/src/index.ts#L308-L327

Of note, what it does or doesn't prefix or transpile may be different to `post-preset-env` since Tailwind v4 requires more modern browser features.

**Comment by stijns96** [comment]
Hi @wongjn,

I understand. But I also checked the [Lightning CSS playground](https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Afalse%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A8585216%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22%40custom-media%20--modern%20(color)%2C%20(hover)%3B%5Cn%5Cn.foo%20%7B%5Cn%20%20background%3A%20yellow%3B%5Cn%5Cn%20%20-webkit-border-radius%3A%202px%3B%5Cn%20%20-moz-border-radius%3A%202px%3B%5Cn%20%20border-radius%3A%202px%3B%5Cn%5Cn%20%20-webkit-transition%3A%20background%20200ms%3B%5Cn%20%20-moz-transition%3A%20background%20200ms%3B%5Cn%20%20transition%3A%20background%20200ms%3B%5Cn%5Cn%20%20%26.bar%20%7B%5Cn%20%20%20%20color%3A%20green%3B%5Cn%20%20%7D%5Cn%7D%5Cn%5Cn%40media%20(--modern)%20and%20(width%20%3E%201024px)%20%7B%5Cn%20%20.a%20%7B%5Cn%20%20%20%20color%3A%20green%3B%5Cn%20%20%7D%5Cn%7D%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D), but that also doesn't seem to work on my end.

**Comment by wongjn** [comment]
For me, the CSS compiles to:

```css
.foo {
  background: #ff0;
  border-radius: 2px;
  transition: background .2s;

  &.bar {
    color: green;
  }
}

@media ((color) or (hover)) and (width > 1024px) {
  .a {
    color: green;
  }
}

```
Which is what I would expect. What is the CSS that you get?

**Comment by stijns96** [comment]
Where do you compile that? On lightning css or somewhere within your project? In my CLI I'm just getting the same output as the input is.

**Comment by wongjn** [comment]
In the Lightning CSS playground you linked.

**Comment by stijns96** [comment]
Yeah, there it's working like expected, but not in the CLI I'm building. I simply pass the tailwind/postcss plugin to my postcss plugins option, but nothing really happens. I would imagine that this would take effect on whatever css file I'm processing, right?

<img width="2214" alt="image" src="https://github.com/user-attachments/assets/920923bb-a7cb-414c-87e3-e9a965cfafce" />

**Comment by wongjn** [comment]
Ensure you have the configuration set up such that Tailwind actually transforms the CSS through Lightning CSS:

https://github.com/tailwindlabs/tailwindcss/blob/608b1399cd66ec391846a81a154a6593db64b040/packages/%40tailwindcss-postcss/src/index.ts#L54

Either set `optimize` to `true` or `process.env.NODE_ENV` to `production`
